src/commands/api/pickDatabase.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AttachedAccountsTreeItem } from 'src/tree/AttachedAccountsTreeItem';
+import { ext } from '../../extensionVariables';
+import { MongoDatabaseTreeItem } from '../../mongo/tree/MongoDatabaseTreeItem';
+import { CosmosDBDatabase } from '../../vscode-cosmosdb.api';
+
+export async function pickDatabase(): Promise<CosmosDBDatabase> {
+    const pickedDatabase = (await ext.tree.showTreeItemPicker([MongoDatabaseTreeItem.contextValue]));
+
+    const attachedAccountsNode = <AttachedAccountsTreeItem | undefined>(await ext.tree.getChildren()).find((subscription) => {
+        return (subscription.id === 'cosmosDBAttachedAccounts');
+    });
+
+    if (pickedDatabase instanceof MongoDatabaseTreeItem) {
+        const databaseFullId = await attachedAccountsNode.attachDatabase(pickedDatabase);
```

Comment:
pickDatabase should not attach anything. It should only pick from existing databases and should have no side effects on the Cosmos DB extension.

On the flip-side I think you should pass in the connectionString to revealTreeItem and _that_ can attach the node if it doesn't find it under the Azure subscriptions

Reply:
I think it's better to create the attachDatabase api command, how do you think?

Reply:
Now reveal command can process any CosmosDB explorer tree items. If we decide to attach the database in it, we will have less functionality and this command won't be general, so I prefer to have two API commands that do simple things.